azure-quantum/azure/quantum/job/job.py

```diff
@@ -142,21 +142,212 @@ def get_results(self, timeout_secs: float = DEFAULT_TIMEOUT):
+
+        payload = self.download_data(self.details.output_data_uri)
+        try:
+            payload = payload.decode("utf8")
+            results = json.loads(payload)
+
+            if self.details.output_data_format == "microsoft.quantum-results.v2":
+                if "DataFormat" not in results or results["DataFormat"] != "microsoft.quantum-results.v2":
+                    raise ValueError(f"\"DataFormat\" was expected to be \"microsoft.quantum-results.v2\" in the Job results for \"{self.details.output_data_format}\" output format.")
+                if "Results" not in results:
+                    raise ValueError(f"\"Results\" field was expected to be in the Job results for \"{self.details.output_data_format}\" output format.")
+                
+                if len(results["Results"]) < 1:
+                    raise ValueError("\"Results\" array was expected to contain at least one item")
+                
+                results = results["Results"]
+
+                if len(results) == 1: 
+                    results = results[0]
+                    if "Histogram" not in results:
+                        raise ValueError(f"\"Histogram\" array was expected to be in the Job results for \"{self.details.output_data_format}\" output format.")
+
+                    histogram_values = results["Histogram"]
+                    outcome_keys = self._process_outcome(histogram_values)
+
+                    # Re-mapping object {'Histogram': [{"Outcome": [0], "Display": '[0]', "Count": 500}, {"Outcome": [1], "Display": '[1]', "Count": 500}]} to {'[0]': {"Outcome": [0], "Count": 500}, '[1]': {"Outcome": [1], "Count": 500}}
+                    return {hist_val["Display"]: {"outcome": outcome, "count": hist_val["Count"]} for outcome, hist_val in zip(outcome_keys, histogram_values)}
+
+                else:
+                    # This is handling the BatchResults edge case
+                    resultsArray = []
+                    for i, result in enumerate(results):
+                        if "Histogram" not in result:
+                            raise ValueError(f"\"Histogram\" array was expected to be in the Job results for result {i} for \"{self.details.output_data_format}\" output format.")
+
+                        histogram_values = result["Histogram"]
+                        outcome_keys = self._process_outcome(histogram_values)
+
+                        # Re-mapping object {'Histogram': [{"Outcome": [0], "Display": '[0]', "Count": 500}, {"Outcome": [1], "Display": '[1]', "Count": 500}]} to {'[0]': {"Outcome": [0], "Count": 500}, '[1]': {"Outcome": [1], "Count": 500}}
+                        resultsArray.append({hist_val["Display"]: {"outcome": outcome, "count": hist_val["Count"]} for outcome, hist_val in zip(outcome_keys, histogram_values)})
+
+                    return resultsArray
+
+            else:
+                raise ValueError(f"This method only handles Jobs which are submitted with the V2 output format.")
```

Comment:
```suggestion
                raise ValueError(f"Getting a results histogram with counts instead of probabilities is not a supported feature for jobs using the \"{self.details.output_data_format}\" output format.")
```

Reply:
Suggestion to hopefully make this error message more user friendly.